themes/finna2/templates/librarycards/selectcard.phtml

```diff
@@ -24,9 +24,26 @@
               if ($target && $targetCount > 1) {
                 $display .= ' (' . $this->transEsc("source_$target", null, $target) . ')';
               }
+              $barcode = $card['barcode'] ?? $username;
+              if ($this->user->getCatUsername() === $card->getCatUsername()) {
+                $profile = $this->ils()->getMyProfile($this->user->toArray());
```

Comment:
Näin ei voi tehdä. getMyProfile tarvitsee parametrikseen patronin, joka saadaan patronLogin-metodilta.

getMyProfile-kutsua ei myöskään voi tehdä joka kerran, kun selectcard.phtml renderöidään. Siitä tulisi ihan tolkuttomasti lisää kuormaa kirjastojärjestelmän puolelle. Tarvitaan välimuisti tms. jotta getMyProfile-kutsut voidaan tehdä kerran. Lisäksi, jos barcode-kenttää ei tule, ei pidä yrittää tehdä viivakoodia. Silloin käyttäjätunnus ei ole sopivan muotoinen, vaan voi olla esim. sähköpostiosoite.